packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -81,19 +104,240 @@ typedef ActionListenerCallback = void Function(Action<Intent> action);
 ///  * [ActionDispatcher], a class that takes an [Action] and invokes it, passing
 ///    a given [Intent].
 abstract class Action<T extends Intent> with Diagnosticable {
+  /// Creates an [Action].
+  Action();
+
+  /// Creates an [Action] that allows itself to be overridden by the closest
+  /// ancestor [Action] in the given [context] that handles the same [Intent],
+  /// if one exists.
+  ///
+  /// When invoked, the resulting [Action] tries to find the closest [Action] in
+  /// the given `context` that handles the same type of [Intent] as the
+  /// `defaultAction`, then calls its [Action.invoke] method. When no override
+  /// [Action]s can be found, it invokes the `defaultAction`.
+  ///
+  /// An overridable action delegates everything to its override if one exists,
+  /// and has the same behavior as its `defaultAction` otherwise. For this
+  /// reason, the override has full control over whether and how an [Intent]
+  /// should be handled, or a key event should be consumed. An override
+  /// [Action]'s [callingAction] property will be set to the [Action] it
+  /// currently overrides, giving it access to the default behavior. See the
+  /// [callingAction] property for an example.
+  ///
+  /// The `context` argument is the [BuildContext] to find the override with. It
+  /// is typically a [BuildContext] above the [Actions] widget that contains
+  /// this overridable [Action].
+  ///
+  /// The `defaultAction` argument is the [Action] to be invoked where there's
+  /// no ancestor [Action]s can't be found in `context` that handle the same
+  /// type of [Intent].
+  ///
+  /// This is useful for providing a set of default [Action]s in a leaf widget
+  /// to allow further overriding, or to allow the [Intent] to propagate to
+  /// parent widgets that also support this [Intent].
+  ///
+  /// {@tool snippet --template=freeform}
```

Comment:
You're going to want to use a different template than `freeform` here, since you don't have a `main` in your sample code (and probably don't really want one). Maybe use the `stateful_widget_material` template? Then you would delete the `SimpleUSPhoneNumberEntry` class, and just have the contents of the  `_SimpleUSPhoneNumberEntryState` in the "``` dart" section (move the `_DeleteDigit` class into the preamble section.

One way you can test this sample is to run:
```
TMPDIR="/tmp" dart dev/bots/analyze_sample_code.dart --temp=samples
```
And it will show you any analyzer errors, and when you've fixed those, then you can grep in the `/tmp/samples` dir for the dart file that has `_DeleteDigit` in it and run that in DartPad to see if it works.

Reply:
I remember the reason this is a `freeform` snippet is that I need the name of the `State` class of the new `StatefulWidget`, as `_DeleteDigit` takes an instance of `_SimpleUSPhoneNumberEntryState`. I'll add a `main` function to this sample.

Reply:
Oh this is a snippet not a sample so no `main` needed.

Reply:
Oh, good point.  In that case you don't need to specify a template at all.

But why not make this into a functional Dartpad sample?  Seems like it's close to that already, and would be a good illustration.